src/setup_node_env/ensure_node_preserve_symlinks.js

```diff
@@ -9,10 +9,50 @@
 (function () {
   var cp = require('child_process');
 
+  var calculateInspectPortOnExecArgv = function (processExecArgv) {
+    var execArgv = [].concat(processExecArgv);
+
+    if (execArgv.length === 0) {
+      return execArgv;
+    }
+
+    var inspectFlagIndex = execArgv.reverse().findIndex(function (flag) {
+      return flag.startsWith('--inspect');
+    });
+
+    if (inspectFlagIndex !== -1) {
+      var inspectFlag;
+      var inspectPortCounter = 9230;
+      var argv = execArgv[inspectFlagIndex];
+
+      if (argv.includes('=')) {
+        // --inspect=port
+        var argvSplit = argv.split('=');
+        var flag = argvSplit[0];
+        var port = argvSplit[1];
+        inspectFlag = flag;
+        inspectPortCounter = Number.parseInt(port, 10) + 1;
+      } else {
+        // --inspect
+        inspectFlag = argv;
+
+        // is number?
+        if (String(execArgv[inspectFlagIndex + 1]).match(/^[0-9]+$/)) {
+          // --inspect port
+          inspectPortCounter = Number.parseInt(execArgv[inspectFlagIndex + 1], 10) + 1;
```

Comment:
Would https://nodejs.org/docs/latest/api/process.html#process_process_debugport be useful here?

Reply:
I do something similar here: https://github.com/elastic/kibana/blob/7fc4fc3a862f0e383cac7a922ccbfb405f694385/src/dev/cli_dev_mode/get_active_inspect_flag.ts

Reply:
I thought about that when I was thinking about how to do this and I started to develop something around it, but then I just found https://github.com/elastic/kibana/blob/master/packages/kbn-optimizer/src/optimizer/observe_worker.ts#L38 and I felt that logic would better fit what I'm trying to accomplish here which is keeping the previous `execArgv` and just update the values 😃

Reply:
👍 They both work

Reply:
I didn't know about `process.debugPort` when I wrote that first implementation though 😉